### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by acunniffe**
Some of the auto generated error messages from Chai are hard to understand. I found a few playing with sweater-comb that really were hard to figure out the right response to. 

This PR (probably one of many the next few weeks) takes those learnings and improves the developer experience

### CODE REVIEW

File: src/rulesets/tests/__snapshots__/headers.test.ts.snap [hunk 1]
```diff
@@ -132,7 +132,7 @@ Object {
       },
       "condition": "be kebab-case",
       "docsLink": "https://github.com/snyk/sweater-comb/blob/main/docs/standards.md#header-field-names",
-      "error": "expected 'bad-header' to equal 'badHeader'",
+      "error": "badHeader is not kebab-case",
```
Comment (acunniffe) [inline src/rulesets/tests/__snapshots__/headers.test.ts.snap:135]: this is subtle, but I think it makes a lot more sense

File: src/rulesets/tests/__snapshots__/lifecycle.test.ts.snap [hunk 2]
```diff
@@ -54,7 +54,7 @@ Object {
       },
       "condition": "not change unless it was wip",
       "docsLink": "https://github.com/snyk/sweater-comb/blob/main/docs/version.md#promoting-stability-of-a-resource-over-time",
-      "error": "expected 'ga' to equal 'beta'",
+      "error": "stability transition from 'ga' to 'beta' not allowed",
```
Comment (acunniffe) [inline src/rulesets/tests/__snapshots__/lifecycle.test.ts.snap:57]: this one actually says what's wrong. A user might read the first and switch back to ga, but will they learn?

File: src/rulesets/tests/__snapshots__/operations.test.ts.snap [hunk 3]
```diff
@@ -103,7 +103,7 @@ Object {
       },
       "condition": "have a summary",
       "docsLink": "https://github.com/snyk/sweater-comb/blob/main/docs/standards.md#operation-summary",
-      "error": "expected undefined to exist",
+      "error": "must have a summary",
```
Comment (acunniffe) [inline src/rulesets/tests/__snapshots__/operations.test.ts.snap:106]: so much more helpful

File: src/rulesets/tests/__snapshots__/operations.test.ts.snap [hunk 4]
```diff
@@ -1894,7 +1894,7 @@ Object {
       },
       "condition": "have operationId",
       "docsLink": "https://github.com/snyk/sweater-comb/blob/main/docs/standards.md#operation-ids",
-      "error": "expected undefined to be truthy",
+      "error": "no operationId provided",
```
Comment (acunniffe) [inline src/rulesets/tests/__snapshots__/operations.test.ts.snap:1897]: so much more helpful

File: src/rulesets/tests/__snapshots__/properties.test.ts.snap [hunk 5]
```diff
@@ -589,7 +589,7 @@ Object {
       },
       "condition": "have snake case keys",
       "docsLink": undefined,
-      "error": "expected false to be truthy",
+      "error": "not-snake-case is not snake-case",
```
Comment (acunniffe) [inline src/rulesets/tests/__snapshots__/properties.test.ts.snap:592]: so much more helpful

**Review by cmars — APPROVED** [review]

**Pull request merged**